fingerprint_verification.py

```diff
@@ -6,39 +6,103 @@
 
 def read_print(filename):
     """Reads the given print and returns a dictionary with all info."""
-    # Your turn!
-
-
-
-
-
-
-# You can add your various check functions below
-
-
-
-
-
-
-
-
-
-
-
-
-
-
-
-
-
-
-
-
+    fingerprint_data = {}
+    with open(filename) as fh:
+        name = fh.readline().strip()
+        width = int(fh.readline().strip())
+        height = int(fh.readline().strip())
+        
+        fingerprint_data['name'] = name
+        fingerprint_data['width'] = width
+        fingerprint_data['height'] = height
+
+        fingerprint = [] 
+
+        for line in filename:
+            cleaned_line = line.rstrip()  # Keep trailing spaces but remove newlines
+        # If the line is empty, create a row of spaces with the correct width
+        if not cleaned_line:
+            cleaned_line = ' ' * width
+        fingerprint.append(cleaned_line)
```

Comment:
I'm pretty sure you wanted all of this inside your `for` loop? Otherwise you are just appending the very last line